### PR TITLE
feat-deferred-deeplinking-support-updated

### PR DESCRIPTION
**Pull request by IloveJavaa**
Added deferred deeplinking support in android
Asana Link - https://app.asana.com/0/0/1203062381831963/f

### CODE REVIEW

File: android/src/main/kotlin/com/example/trackierfluttersdk/TrackierfluttersdkPlugin.kt [hunk 1]
```diff
@@ -89,9 +89,25 @@ class TrackierfluttersdkPlugin : FlutterPlugin, MethodCallHandler {
...
         trackierSDKConfig = TrackierSDKConfig(context, appToken, environment)
-        trackierSDKConfig.setSDKVersion("1.6.29")
+        trackierSDKConfig.setSDKVersion("1.6.30")
         trackierSDKConfig.setSDKType("flutter_sdk")
         trackierSDKConfig.setAppSecret(secretId, secretKey)
+    
+        if (configMap.containsKey("deeplinkCallbacks")) {
+            val dartMethodName = configMap["deeplinkCallbacks"] as String?
+            if (dartMethodName != null) {
+                if (channel != null) {
```
Comment (Hemant-Mann) [inline android/src/main/kotlin/com/example/trackierfluttersdk/TrackierfluttersdkPlugin.kt:99]: merge both if blocks
if dartMethodName != null && channel != null

File: lib/trackierconfig.dart [hunk 2]
```diff
@@ -1,27 +1,55 @@
...
+          case _deferredDeeplinkCallbackName:
+            if (deferredDeeplinkCallback != null) {
+              String? uri = call.arguments['uri'];
+              if (deferredDeeplinkCallback != null) {
+                deferredDeeplinkCallback!(uri);
+              }
+            }
+            break;
+        }
+      }catch (e) {
```
Comment (Hemant-Mann) [inline lib/trackierconfig.dart:40]:
```suggestion
      } catch (e) {
```

File: lib/trackierconfig.dart [hunk 3]
```diff
@@ -1,27 +1,55 @@
...
   }
 
+  void _initCallbackHandlers() {
+    _channel.setMethodCallHandler((MethodCall call) async {
+      try {
+        switch (call.method) {
+          case _deferredDeeplinkCallbackName:
+            if (deferredDeeplinkCallback != null) {
+              String? uri = call.arguments['uri'];
+              if (deferredDeeplinkCallback != null) {
```
Comment (Hemant-Mann) [inline lib/trackierconfig.dart:34]: this check is already added in the above if block why are we adding it twice?

File: lib/trackierconfig.dart [hunk 4]
```diff
@@ -1,27 +1,55 @@
...
+    });
+  }
+
   Map<String, String?> get toMap {
     Map<String, String?> configMap = {
       'appToken': appToken,
       'environment': envirnoment,
       'secretId': secretId,
       'secretKey': secretKey,
+      'deeplinkCallbacks' : _deferredDeeplinkCallbackName,
```
Comment (Hemant-Mann) [inline lib/trackierconfig.dart:52]:
```suggestion
      'deeplinkCallback' : _deferredDeeplinkCallbackName,
```

File: android/src/main/kotlin/com/example/trackierfluttersdk/TrackierfluttersdkPlugin.kt [hunk 5]
```diff
@@ -89,9 +89,25 @@ class TrackierfluttersdkPlugin : FlutterPlugin, MethodCallHandler {
             environment = configMap.get("environment") as String
         }
         trackierSDKConfig = TrackierSDKConfig(context, appToken, environment)
-        trackierSDKConfig.setSDKVersion("1.6.29")
+        trackierSDKConfig.setSDKVersion("1.6.30")
         trackierSDKConfig.setSDKType("flutter_sdk")
         trackierSDKConfig.setAppSecret(secretId, secretKey)
+    
+        if (configMap.containsKey("deeplinkCallbacks")) {
```
Comment (Hemant-Mann) [inline android/src/main/kotlin/com/example/trackierfluttersdk/TrackierfluttersdkPlugin.kt:96]:
```suggestion
        if (configMap.containsKey("deeplinkCallback")) {
```

**Review by Hemant-Mann — COMMENTED** [review]
- Also update CHANGELOG.md file since we are bumping the version @IloveJavaa

File: lib/trackierconfig.dart (hunk 2 above)
Comment (IloveJavaa) [reply lib/trackierconfig.dart:40]: Done

File: lib/trackierconfig.dart (hunk 3 above)
Comment (IloveJavaa) [reply lib/trackierconfig.dart:34]: Adding just like Adjust. But i have removed and merge in the first check

File: lib/trackierconfig.dart (hunk 4 above)
Comment (IloveJavaa) [reply lib/trackierconfig.dart:52]: Done

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,3 @@
-## 1.6.29
+## 1.6.30
 
-- feat: additional fields in events
-  Add support for user name
-  Support for user phone
-  Event class now supports coupon code and discount
+- feat: deferred deeplinking support
```
Comment (Hemant-Mann) [inline CHANGELOG.md:3]:
```suggestion
- feat: deferred deeplinking support for android
```

**Review by Hemant-Mann — APPROVED** [review]

**Pull request merged**